examples/seismic/tutorials/04_scipy_optimise.ipynb

```diff
@@ -0,0 +1,397 @@
+    "    src.coordinates.data[0, :] = [30, param['shot_id']*1000./(param['nshots']-1)]\n",
+    "\n",
+    "    # Set up receiver data and geometry.\n",
+    "    nreceivers = 101  # Number of receiver locations per shot\n",
+    "    rec = Receiver(name='rec', grid=true_model.grid, npoint=nreceivers, ntime=nt)\n",
+    "    rec.coordinates.data[:, 1] = np.linspace(0, true_model.domain_size[0], num=nreceivers)\n",
+    "    rec.coordinates.data[:, 0] = 980. # 20m from the right end\n",
+    "\n",
+    "    # Set up solver - using model_in so that we have the same dt,\n",
+    "    # otherwise we should use pandas to resample the time series data. \n",
+    "    solver = AcousticWaveSolver(true_model, src, rec, space_order=4)\n",
+    "\n",
+    "    # Generate synthetic receiver data from true model\n",
+    "    true_d, _, _ = solver.forward(src=src, m=true_model.m)\n",
+    "\n",
+    "    return src, true_d, nt, solver"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create operators for gradient based inversion\n",
+    "To perform the inversion we are going to use [scipy.optimize.minimize(method=’L-BFGS-B’)](https://docs.scipy.org/doc/scipy/reference/optimize.minimize-lbfgsb.html#optimize-minimize-lbfgsb).\n",
+    "\n",
+    "First we define the functional, ```f```, and gradient, ```g```, operator (i.e. the function ```fun```) for a single shot of data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from devito import Function, clear_cache\n",
+    "\n",
+    "# Create FWI gradient kernel for a single shot\n",
+    "def fwi_gradient_i(x, param):\n",
+    "    # Need to clear the workers cache.\n",
+    "    clear_cache()\n",
+    "\n",
+    "    # Get the current model and the shot data for this worker.\n",
+    "    model0 = get_initial_model()\n",
+    "    model0.m.data[:] = x.astype(np.float32).reshape(model0.m.data.shape)\n",
+    "    src, rec, nt, solver = get_data(param)\n",
+    "    \n",
+    "    # Create symbols to hold the gradient and the mismatch between\n",
+    "    # the 'measured' and simulated data.\n",
+    "    grad = Function(name=\"grad\", grid=model0.grid)\n",
+    "    residual = Receiver(name='rec', grid=model0.grid, ntime=nt, coordinates=rec.coordinates.data)\n",
+    "    \n",
+    "    # Compute simulated data and full forward wavefield u0\n",
+    "    d, u0, _ = solver.forward(src=src, m=model0.m, save=True)\n",
+    "        \n",
+    "    # Compute the data mismatch (residual) and objective function  \n",
+    "    residual.data[:] = d.data[:] - rec.data[:]\n",
+    "    f = .5*np.linalg.norm(residual.data.flatten())**2\n",
+    "    \n",
+    "    # Compute gradient using the adjoint-state method. Note, this\n",
+    "    # backpropogates the data mismatch through the model.\n",
```

Comment:
The word "backpropagate" means something very specific for ANN folk. That's not what is happening here. I would use more neutral language here, like "propagate backwards".

Reply:
>That's not what is happening here.

This is the correct wording. It is called back-propagation.

Reply:
Agreed. Corrected spelling.